spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java

```diff
@@ -152,4 +174,45 @@ public String manifest(@PathVariable("name") String name, @PathVariable("version
 	public Collection<Deployer> platformList() {
 		return this.streamService.platformList();
 	}
+
+	/**
+	 * {@link org.springframework.hateoas.ResourceAssembler} implementation that converts
+	 * {@link StreamDeployment}s to {@link StreamDeploymentResource}s.
+	 */
+	class Assembler extends ResourceAssemblerSupport<StreamDeployment, StreamDeploymentResource> {
+
+		private final StreamDefinition streamDefinition;
+
+		public Assembler(StreamDefinition streamDefinition) {
```

Comment:
I feel like, we should be passing here only the pre-computed properties rather than an entire domain class?